core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -333,7 +333,12 @@ public Object visitInvalidateL1Command(InvocationContext ctx, InvalidateL1Comman
 
    @Override
    public final Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) {
-      return setSkipRemoteGetsAndInvokeNextForDataCommand(ctx, command, wrapEntryIfNeeded(ctx, command));
+      boolean isEntryVisited = command.loadType() != VisitableCommand.LoadType.DONT_LOAD;
```

Comment:
Sorry can you just explain what happened here?

Reply:
Previously we did a get + put for a putIfAbsent command, this combination issues a CacheEntryVisited because of the get. With the changes we added, we stopped issuing the event. I did this so we still issue the event when the entry is visited in the putIfAbsent.

Reply:
Reverted this stuff and removed the test check.